### ISSUE TITLE
Option to hide sites and categories from camilt

### ISSUE THREAD

**Issue by dobbelina** [dobbelina/repository.dobbelina#327]
I have been forced to decline an update from camilt because of an
conflict with an old fork.
I have sent a pull request to him/her to update the fork before making any
changes.
Camilt added the option to hide categories and sites in the settings which was real
nice, however it will mean that additional editing is necessary when updating the addon.
Here's a test version that i downloaded from his/her fork.
Please tell me what you think dear contributors, and big thanks to Camilt
for the effort!
[Test-Version.zip](https://github.com/dobbelina/repository.dobbelina/files/5558551/Test-Version.zip)

**Comment by dobbelina** [comment]
If feasible a Favorites Sites section in addition to the already existing Favorites would be better i think.
That way one would only need to add sites instead of deselecting them in the settings.

Just an idea :bulb:

**Comment by flapbird1** [comment]
Hi dobbelina,
IMHO, I'd focus all mayor development updating UWC to Kodi 19 / Python 3, as Beta 1 is already out.
Thanks to all developers for the great work

**Comment by jdoedev123** [comment]
Nicely done. I was considering a similar method of settings to hide websites.
Presently running 3 forks of UWC. The semi-official 1.2.32, my personal development version based on 1.2.28 and Cammit's based on 1.2.29.
It was tricky to get them to install and run concurrently by editing addon.xml to give different versions and plugin names. Settings differences required editing utils.py to change the __scriptid__  in line 23 to different names consistent with the plugin namespaces. This is a dev and test problem only.

**Comment by jdoedev123** [comment]
Flapbird1, moving to Kodi 19 Python 3 etc is a can of worms I have been putting off. As mentioned above I am already running multiple forks of UWC and Kodi 18 versions in containers but have not found a suitable containerized Kodi 19. I could build a containerized version from source (I have built Kodi from source for Linux and Android) but it is a pain in the ...
Eventually

**Comment by WhitecreamDev** [comment]
> 
> 
> Nicely done. I was considering a similar method of settings to hide websites.
> Presently running 3 forks of UWC. The semi-official 1.2.33, my personal development version based on 1.2.28 and Cammit's based on 1.2.29.
> It was tricky to get them to install and run concurrently by editing addon.xml to give different versions and plugin names. Settings differences required editing utils.py to change the **scriptid** in line 23 to different names consistent with the plugin namespaces. This is a dev and test problem only.

Actually, you won't need the scriptid, which is used on line 72.
Just change it to: `addon = xbmcaddon.Addon()`
The id in Addon() is optional, and defaults standard to id from addon.xml
So less editing for different UWC versions, only addon.xml and foldername :)

Also, you can 'containerize' Kodi using Portable Mode: https://kodi.wiki/view/Portable_mode
Copy your kodi user folder, and install 19.
Copy the kodi 19 program folder to a separate folder/drive.
Reinstall Kodi 18, copy the user folder back.
Run Kodi19 with 'kodi.exe -p'

**Comment by jdoedev123** [comment]
Thanks, WhiteCreamDev you are the OG of UWC. I develop on Linux which does not support Portable Mode. As a Linux OG I could probably figure out my own work around but I am trying to use more modern tools these days to isolate dev and runtime environments. In the old days I used to carry multiple laptops and mini PCs around the world to isolate environments.

**Comment by Gujal00** [comment]
@jdoedev123 Have a look at this
[code]https://github.com/ehough/docker-kodi[/code]
You can run kodi in a docker container, so you could run as many versions you want as individual containers :)

**Comment by camilt** [comment]
Check the new method of enable/disabel sites/categories in #343 
It uses contextual menu for disabling, then creates in Add-on settings an category for disabled sites. From there you can reEnable them. Please test it and send feedback

**Comment by dobbelina** [comment]
Thanks for the addition camilt!
I will roll it out immediately.

**Comment by flapbird1** [comment]
Awesome Camilt!
Keep up the great work

**Comment by imurreflection** [comment]
This is amazing @camilt
You are really a coding guru.

**Comment by dobbelina** [comment]
Having an odd issue where ParadiseHill have disappeared.
I wonder if this change has anything to do with it #343
Issue is here:https://github.com/dobbelina/repository.dobbelina/issues/344#issuecomment-735297139

**Comment by jdoedev123** [comment]
@Gujal00  Yes I found that some time ago. Finally had some time to mess with the Docker code and produce my own Kodi 19 dock container. It took some customization to get a custom Kodi container to run with graphics.
To get UWC to at least load I had to get the latest resolveurl and resolveurl.xxx from github and modify FmProxy. Eventually I loaded UWC 1.2.34 with a modified addon.xml. it did not go well at first.
Rather than use a tool to convert Python 2 to Python 3 I manually fixed each bug as it was thrown until I finally have a first draft at a working UWC in Kodi 19. Played my first video...
Will probably restart with Python 2 to Python 3 converter but at least I know where most of the incompatibilities are. Most are Python 3 differences with URLLIB  and how Strings are handled but a few Kodi differences caused problems as well.
